types/browser-sync/index.d.ts

```diff
@@ -291,9 +291,10 @@ declare namespace browserSync {
     interface ProxyOptions {
         target?: string;
         middleware?: MiddlewareHandler;
-        ws: boolean;
-        reqHeaders: (config: any) => Hash<any>;
-        proxyRes: (res: http.ServerResponse, req: http.IncomingMessage, next: Function) => any;
+        ws?: boolean;
+        reqHeaders?: (config: any) => Hash<any>;
+        proxyRes?: (res: http.ServerResponse, req: http.IncomingMessage, next: Function) => any;
+        proxyReq?: (res: http.ServerRequest) => any;
```

Comment:
- can you update the doc comment above line 52?
- can you add some tests? for example, the ones from the [docs](https://browsersync.io/docs/options#option-proxy) should suffice.